### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by pathouse**

### CODE REVIEW

File: app/models/user.go [hunk 1]
```diff
@@ -1,25 +1,36 @@
...
 }
 
-func (u *User) LinkTitle() string {
-	return "User Link Title"
+func (u *User) GenerateHashedPassword(password string) error {
+	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
+		return err
+	}
+	u.PasswordHash = string(hash[:])
```
Comment (Peleg) [inline app/models/user.go:26]: whats this `hash[:]` about?

File: app/controllers/sessions.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	s.Validation.Required(password)
+	s.Validation.Required(passwordConfirmation)
+	s.Validation.MinSize(password, 8)
+	s.Validation.Match(password, regexp.MustCompile(passwordConfirmation))
+
+	if s.Validation.HasErrors() {
+		s.Validation.Keep()
+		s.FlashParams()
+		return s.Redirect(routes.Sessions.Register())
+	}
```
Comment (Peleg) [inline app/controllers/sessions.go:34]: @pathouse ive seen how this is done in the revel side also in the controller, so thats prolly the convention, but do you think itll be better to have it in the model as `func Save` or something and have it return the errors?

Comment (Peleg) [inline app/controllers/sessions.go:34]: ah nvm, then we wont have the session. guess fat controllers, lean models :P

File: app/models/user.go (hunk 1 above)
Comment (pathouse) [inline app/models/user.go:26]: hash is an array so converting the slice of the whole thing to a string. tbh not entirely clear on why this needs to be done but it does as far as i know

File: app/controllers/sessions.go (hunk 2 above)
Comment (pathouse) [inline app/controllers/sessions.go:34]: yeah for now i found it easier to use revel's baked in validation, in the future i'd like to move it out though I definitely agree with you there

**Pull request closed without merging**